wallet/listeners/transaction_listener.go

```diff
@@ -226,12 +226,14 @@ func (l *TransactionListener) processSalePayment(txid string, output wallet.Tran
 		}
 
 		// update divisibility from contract listing
-		customDivisibility := currencyDivisibilityFromContract(l.multiwallet, contract)
-		if customDivisibility != currencyValue.Currency.Divisibility {
-			currencyValue.Currency.Divisibility = customDivisibility
-			if err := currencyValue.Valid(); err != nil {
-				log.Errorf("Invalid currency divisibility (%d) found in contract (%s): %s", customDivisibility, orderId, err.Error())
-				return
+		if contract.VendorListings[0].Metadata.ContractType != pb.Listing_Metadata_CRYPTOCURRENCY {
+			customDivisibility := currencyDivisibilityFromContract(l.multiwallet, contract)
+			if customDivisibility != currencyValue.Currency.Divisibility {
+				currencyValue.Currency.Divisibility = customDivisibility
+				if err := currencyValue.Valid(); err != nil {
+					log.Errorf("Invalid currency divisibility (%d) found in contract (%s): %s", customDivisibility, orderId, err.Error())
+					return
+				}
```

Comment:
Any reason we wouldn't have `currencyDivisibilityFromContract` always do the right thing no matter what contract we gave it?

Reply:
What would be the return in the case of a cryptocurrency listing? An error? I suppose we should check the listing in the function and return an error, though I don't see a reason to call a function you know will return an error in the crypto listing case.

Reply:
> I don't see a reason to call a function you know will return an error in the crypto listing case.
That's good point. Disregard then.